parser.y

```diff
@@ -6120,6 +6120,10 @@ RoleNameString:
 	{
 		$$ = $1
 	}
+|	"ROLE"
```

Comment:
It is very odd we special-case `ROLE` here, given that other unreserved keywords could also be used, e.g.

```sql
CREATE ROLE ROLLBACK;
CREATE ROLE ADMIN;
CREATE ROLE COPY;
```

I suggest we just change the `identifier` on Line 6119 to `Identifier`.

(Please also add test cases for these 3 statements.)

Reply:
ok

Reply:
@kennytm using `Identifier` will cause conflict....

Reply:
If it is a shift/reduce conflict it could be resolved by specifying a precedence.

Reply:
it is a reduce/reduce conflict, because `NONE` is an unreserved keyword.

Reply:
🤔 

So we need to exclude the unreserved keywords `EXECUTE`, `SUPER`, `PROCESS`, `RELOAD`, `EVENT` (for reduce/reduce on `GrantStmt`/`GrantRoleStmt`) and `NONE` (for reduce/reduce on `SetDefaultRoleOpt`). 

This would mean we should separate these six tokens from `UnReservedKeyword` into another list...